lib/_http_client.js

```diff
@@ -32,9 +32,17 @@ function ClientRequest(options, cb) {
   var defaultAgent = options._defaultAgent || Agent.globalAgent;
   if (agent === false) {
     agent = new defaultAgent.constructor();
-  } else if ((agent === null || agent === undefined) &&
-             typeof options.createConnection !== 'function') {
-    agent = defaultAgent;
+  } else if (agent === null || agent === undefined) {
+    if (typeof options.createConnection !== 'function') {
+      agent = defaultAgent;
+    }
+    // Explicitly pass through this statement as agent will not be used
+    // when createConnection is provided.
+  } else if (!(agent instanceof Agent.Agent)) {
+    throw new TypeError(
+       'Agent option must be an instance of http.Agent, undefined or false. ' +
+       `Received type: ${typeof agent} instead`
```

Comment:
@jasnell - sorry for the delay. Addressed your feedback and removed the extra bits.